### PR TITLE
Enabled full access to Filesystem lib and dstream lib (4-2-stable)

### PR DESCRIPTION
**Pull request by korydraughn**
Ported from #6845 

All unit tests passed.

Running core tests.

### CODE REVIEW

**Comment by korydraughn** [comment]
All core tests passed.

What's even more amazing is that bumping the executors up to 6 instead of 4 resulted in the following:
```bash
-----
All tests passed! :)
time elapsed: [5870.4971]seconds ([   1]hours [37.8416]minutes)
==== end of test run results ====
```

**Review by alanking — APPROVED** [review]

**Comment by korydraughn** [comment]
Added #'s

File: plugins/microservices/msi_tests/msi_assertion_macros.hpp [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+// NOLINTNEXTLINE(cppcoreguidelines-macro-usage)
+#define IRODS_MSI_ASSERT(x)                                                    \
+  if (x) {                                                                     \
+    rodsLog(LOG_NOTICE, "ASSERTION PASSED [{}:{}]: " #x, __FILE__, __LINE__);  \
+  }                                                                            \
+  else {                                                                       \
+    rodsLog(LOG_ERROR, "ASSERTION FAILED [{}:{}]: " #x, __FILE__, __LINE__);   \
+    return 1;                                                                  \
+  }
```
Comment (korydraughn) [inline plugins/microservices/msi_tests/msi_assertion_macros.hpp:36]: Hold please.

Comment (alanking) [reply plugins/microservices/msi_tests/msi_assertion_macros.hpp:36]: Dang. This is why I have trust issues :p

Comment (korydraughn) [reply plugins/microservices/msi_tests/msi_assertion_macros.hpp]: Fixed it!

**Comment by korydraughn** [comment]
It's ready for merging now.

**Review by alanking — APPROVED** [review]

**Pull request merged**